### PR TITLE
Add release branch policy

### LINKED ISSUE #1002: Release branch management
We manoeuvred ourselves into a tough spot with our current branch management for releases and pre-releases. As it was initially not foreseen that we will have a long-running pre-release branch due to some issues with the cabal plugin, we did not split off a pre-release branch but tried to manage everything in master.

That tripped us up in the latest release as it makes it rather awkward to make releases for the current stable release of the vscode plugin.

Thus, I propose a new branch management policy:

* master is always the stable release 
* pre-releases are performed from the branch `release/<ver-num>`.

### PR DESCRIPTION
**Pull request by fendor**
closes #1002

### CODE REVIEW

File: docs/Release.md [hunk 1]
```diff
@@ -16,6 +16,19 @@ Follow this list for items that must be completed for release of the `vscode-has
...
+
+Sometimes there is a release (stable) and pre-release (unstable) at the same time and we need to do a release for the stable release and sometimes we need to do a release for the pre-release series.
+To simplify the release management, the following policy is in place:
+
+* The branch `master` contains the current stable release
+  * As such, its `package.json` must always have the form `major.EVEN_NUMBER.patch`
+  * Dependency version bumps are automatically performed by dependabot against `master`
+  * For each release, a tag must be created
+* Pre-releases are located on a separate branch called `pre-release-<major.ODD_NUMBER>`
+  * Before a release, the branch is rebased on top of current master
```
Comment (hasufell) [inline docs/Release.md:29]: Are you implying we should force push to prerelease branches?

**Review by hasufell — APPROVED** [review]

File: docs/Release.md (hunk 1 above)
Comment (fendor) [reply docs/Release.md:29]: Yeah, I am implying that, but when you say it like this, this seems like a terrible idea. 
Maybe a merge commit is a better idea?

File: docs/Release.md [hunk 2]
```diff
@@ -16,6 +16,19 @@ Follow this list for items that must be completed for release of the `vscode-has
   - Github actions will automatically release the extension to VSCode- and VSX-Marketplace.
   - If you want to create a pre-release, create a [pre-release in Github](https://github.com/haskell/vscode-haskell/releases). The github action will perform the appropriate actions automatically and publish the pre-release of the extension to VSCode- and VSX-Marketplace.
 
+## Branching policy
+
+Sometimes there is a release (stable) and pre-release (unstable) at the same time and we need to do a release for the stable release and sometimes we need to do a release for the pre-release series.
+To simplify the release management, the following policy is in place:
+
+* The branch `master` contains the current stable release
```
Comment (michaelpj) [inline docs/Release.md:24]: Perhaps I don't understand how this works here, but isn't the usual approach that master is always the "development" branch (which looks more like a pre-release) and you make an explicit release branch for stable releases? Is this due to some magic where we tell vscode to follow a branch for new releases or something?

Comment (fendor) [reply docs/Release.md:24]: There is no magic involved. So far, we didn't choose a particular branching/release strategy and master was equal to develop, and at the same time the release branch. Development is slow, but currently we have two branches that require maintenance, hence a policy change. Are you proposing that master becomes the pre-release branch, and we add a branch, say `release/2.4`, where we add commits which are backported from master?

The only thing to keep in mind, https://marketplace.visualstudio.com/items?itemName=haskell.haskell shows the README from the master branch, afaict.

Comment (michaelpj) [reply docs/Release.md:24]: Yeah, that's what I'm suggesting. It's what I've seen in most other software projects, but :shrug: 

Normally what you do is that you decide you're cutting a release, and then you make a release branch to which you backport stuff if you need to, and then you eventually tag off the release branch.

Comment (fendor) [reply docs/Release.md]: The policy has been updated, thanks for your input!

**Pull request merged**